_drafts/gdbundle.md

```diff
@@ -0,0 +1,549 @@
+the activated virtual environment, Conda environment, or non-default Python
+installation.
+
+The second part, `gdbundle.init()`, is what is new and will load installed
+gdbundle plugins.
+
+> NOTE: The major (and ideally minor) version of Python linked with GDB should
+> match the version that is activated at the time of launch, whether that is a
+> virtual environment, Conda environment, `brew` Python installation, etc.
+> Mismatched major versions will cause packages and plugins to either break in
+> mysterious ways or not show up at all. More information can be found in the
+> [gdbundle README.md](https://github.com/memfault/gdbundle)
+
+## gdbundle Example Plugin
+
+No plugin manager introduction would be complete without an example plugin. I've
+created a "Hello World" plugin for gdbundle called
+[`gdbundle-example`](https://github.com/memfault/gdbundle-example). Let's dig in
+and see how it's built.
+
+### Structure
+
+Below is the directory structure of our example plugin that works with GDB and
+LLDB.
+
+```
+├── README.md
+├── gdbundle_example
+│   ├── __init__.py
+│   ├── gdb_loader.py
+│   ├── lldb_loader.py
+│   └── scripts
+│       ├── example_gdb.gdb
+│       ├── example_gdb.py
+│       └── example_lldb.py
+└── pyproject.toml
+```
+
+Simple, right? I've opted to use `pyproject.toml` instead of `setup.py` because
+I wanted to embrace the future of standardized packaging, but note that these
+packages will work with Python 2 and 3, so it shouldn't matter.
+
+### How It Works
+
+Let's look at the code in `gdbundle_example/gdb_loader.py`
+
+```python
+import gdb
+import os
+
+PACKAGE_DIR = os.path.dirname(__file__)
+
+SCRIPT_PATHS = [
+    [PACKAGE_DIR, 'scripts', 'example_gdb.gdb'],
+    [PACKAGE_DIR, 'scripts', 'example_gdb.py']
+]
+
+def _abs_path(path):
+    return os.path.abspath(os.path.join(*path))
+
+def gdbundle_load():
+    for script_path in SCRIPT_PATHS:
+        gdb.execute("source {}".format(_abs_path(script_path)))
+```
+
+Notice the `gdbundle_load` function which `gdbundle` will call directly. When
+this is called, it takes each script file located in `scripts/` and tells GDB to
+`source` it. Notice that both GDB Python and GDB Command scripts work!
+
+By using `gdb.execute("source <script>")`, plugins can source GDB scripts that
+were never designed with gdbundle in mind or designed to be distributed.
+
+## A Few More Examples
+
+Given the flexibility of gdbundle itself, it can allow developer to package up
+any of the already existing GDB scripts and make installation easier for
+everyone.
+
+Here are a few more examples of plugins I've created that wrap popular packages
+that I use frequently.
+
+- [gdbundle-gdb-dashboard](https://github.com/memfault/gdbundle-gdb-dashboard) -
+  gdbundle plugin for
+  [cyrus-and/gdb-dashboard](https://github.com/cyrus-and/gdb-dashboard).
+- [gdbundle-PyCortexMDebug](https://github.com/memfault/gdbundle-PyCortexMDebug) -
```

Comment:
I forget if this package still does the svd parse on import but that's something I would want to source conditionally if so hehe

Reply:
Haha, it doesn't do the svd parse on load since an svd file isn't given as an argument. But yeah, it takes forever!